Mathlib/Analysis/InnerProductSpace/Basic.lean

```diff
@@ -708,6 +842,51 @@ theorem real_inner_mul_inner_self_le (x y : F) : ⟪x, y⟫_ℝ * ⟪x, y⟫_ℝ
 
 #align real_inner_mul_inner_self_le real_inner_mul_inner_self_le
 
+end PreBasicProperties
+
+section BasicProperties
+
+variable [NormedAddCommGroup E] [InnerProductSpace 𝕜 E]
+variable [NormedAddCommGroup F] [InnerProductSpace ℝ F]
+
+local notation "⟪" x ", " y "⟫" => @inner 𝕜 _ _ x y
+
+local notation "IK" => @RCLike.I 𝕜 _
+
+local postfix:90 "†" => starRingEnd _
+
+export InnerProductSpace (norm_sq_eq_inner)
+
+@[simp]
+theorem inner_self_eq_zero {x : E} : ⟪x, x⟫ = 0 ↔ x = 0 := by
+  rw [inner_self_eq_norm_sq_to_K, sq_eq_zero_iff, ofReal_eq_zero, norm_eq_zero]
+#align inner_self_eq_zero inner_self_eq_zero
+
+theorem inner_self_ne_zero {x : E} : ⟪x, x⟫ ≠ 0 ↔ x ≠ 0 :=
+  inner_self_eq_zero.not
+#align inner_self_ne_zero inner_self_ne_zero
+
+variable (𝕜)
+
+theorem ext_inner_left {x y : E} (h : ∀ v, ⟪v, x⟫ = ⟪v, y⟫) : x = y := by
+  rw [← sub_eq_zero, ← @inner_self_eq_zero 𝕜, inner_sub_right, sub_eq_zero, h (x - y)]
+#align ext_inner_left ext_inner_left
+
+theorem ext_inner_right {x y : E} (h : ∀ v, ⟪x, v⟫ = ⟪y, v⟫) : x = y := by
+  rw [← sub_eq_zero, ← @inner_self_eq_zero 𝕜, inner_sub_left, sub_eq_zero, h (x - y)]
+#align ext_inner_right ext_inner_right
+
+@[simp]
+theorem inner_self_nonpos {x : E} : re ⟪x, x⟫ ≤ 0 ↔ x = 0 := by
+  rw [← norm_sq_eq_inner, (sq_nonneg _).le_iff_eq, sq_eq_zero_iff, norm_eq_zero]
+#align inner_self_nonpos inner_self_nonpos
```

Comment:
Unless I missed something `𝕜` is now explicit here. Can we change it back for consistency?

Reply:
sorry that I missed this, yes.